docs/hub/adding-a-model.md

```diff
@@ -85,35 +180,123 @@ git add .
+## Installing the library and logging in
 
-8. Add TensorBoard traces
+Before we begin, you should make sure the `huggingface_hub` library is installed on your system by running the following `bash` command:
 
-Models trained with 🤗 Transformers will generate [TensorBoard traces](https://huggingface.co/transformers/main_classes/callback.html?highlight=tensorboard#transformers.integrations.TensorBoardCallback) by default if [`tensorboard`](https://pypi.org/project/tensorboard/) is installed.
+>>> pip install huggingface_hub
 
-Beyond 🤗 Transformers, any repository that contains TensorBoard traces (filenames that contain `tfevents`) is categorized with the [`TensorBoard` tag](https://huggingface.co/models?filter=tensorboard). As a convention we suggest that you save traces under the `runs/` subfolder. The "Training metrics" tab then makes it easy to review charts of the logged variables, like the loss or the accuracy.
+Afterwards, you should login with your credentials. To find your credentials you can go to your [settings](https://huggingface.co/settings/token) from the Hugging Face website and copy the current token there.
+
+To login to your profile, there are two options:
+- Logging in from a Jupyter Notebook that has ipywidgets enabled
+- Logging in from the command line
+
+To login through Jupyter, run the following inside of a Notebook cell:
+
+>>> from huggingface_hub import notebook_login
+>>> notebook_login()
+
+You will be presented with a prompt similar to the following, where it will ask you to paste in that login token from earlier
+
+![/docs/assets/hub/notebook_login.png](/docs/assets/hub/notebook_login.png)
+
+To login through the command line, run the following from a terminal:
+
+>>> huggingface-cli login
+
+You will be presented with a prompt asking for you to paste your token
+![/docs/assets/hub/cli-login.png](/docs/assets/hub/cli-login.png)
+
+After either login method is chosen, you will be asked to run:
+>>> git config --global credential.helper store
+
+This ensures that git is looking at our newly-stored credentials any time we wish to push to the Hub
+
+> Note: You may find that `HfApi` has a `set_access_token` function. This does not set all the permissions needed at each location, and is more for internal use. You should use one of the two methods mentioned above.
+
+## Creating a repository
+
+When using the `huggingface_hub`, we can create a new repository from just a few lines of code!
+First we need to instantiate the `HfApi` class, which holds all of the magic:
+>>> from huggingface_hub import HfApi
+>>> api = HfApi()
+
+Afterwards we can run the `create_repo` function, specifying a number of settings and options for our new repository:
+>>> api.create_repo(
+>>>   name = "dummy", # The name of our repository
+>>>   organization = None, # The username of the expected repository. Automatically grabs your logged-in profile name
```

Comment:
I don't know if "username" is the right word here. "namespace" might be a better fit